TOOLS/docutils-wrapper.py

```diff
@@ -56,10 +56,12 @@ def remove(path):
 
 try:
     proc = subprocess.run(argv, check=True)
-    convert_depfile(output, depfile)
+    if depfile is not None:
+        convert_depfile(output, depfile)
 except:
     remove(output)
-    remove(depfile)
+    if depfile is not None:
+        remove(depfile)
```

Comment:
Doesn't the try/except from the previous commit handle this one already?

Reply:
No, if `None` is passed:
```
TypeError: remove: path should be string, bytes or os.PathLike, not NoneType
```
We could also catch the TypeError but that looks weird.

Reply:
That doesn't count as an exception? Weird, well whatever I guess.

Reply:
I like to be explicit.

Exceptions for things that would be prone to TOCTU errors; the `FileNotFound`.
Checks for things that can be checked; the `None`.

Reply:
In that case, you should probably add OSError to that one as well. Not sure if there are any other ones.